src/hdmf_zarr/backend.py

```diff
@@ -1202,7 +1261,11 @@ def read_builder(self):
         return f_builder
 
     def __set_built(self, zarr_obj, builder):
-        fpath = zarr_obj.store.path
+        # fpath = zarr_obj.store.path
+        if isinstance(zarr_obj.store, zarr.storage.ConsolidatedMetadataStore):
+            fpath = zarr_obj.store.store.path
+        else:
+            fpath = zarr_obj.store.path
```

Comment:
Since you use this logic of having to distinguish by the the storage type to determine the path, I think it would be useful to make a small private helper function for this, e.g,. ``__get_store_path``

Reply:
Agreed. This was a checkpoint push since I got everything to pass. I need to optimize and finish some tests.